Track OSIE version information

ScottGarman opened the pull request:
Normally I'd create a file called VERSION that would get packaged up and could be read from, but since we're generating multiple containers, ensuring access to this file gets a bit complicated. So I thought it would be easier to template some variables in specific OSIE scripts that could be substituted at build time.

Consider this an RFC pull request and if we can get consensus on the technical approach for this, there are several more areas where we can make use of this version information - e.g, sending it to hegel on initial connection.

ScottGarman committed:
Pass git version and drone build information to OSIE containers

This sets some variables in the Makefile to track the OSIE git version
sha, branch, and Drone build number at build time, and passes them along
to the OSIE and osie-runner containers using ARG and ENV docker
variables. These variables are then available to the OSIE scripts and
can be referenced at will.

Signed-off-by: Scott Garman <sgarman@packet.com>

mikemrm commented:
What do you think about just including it as args in the dockerfile and creating a .env file during build time that can just get sourced?

ScottGarman commented:
@mikemrm That's an interesting idea. I think I like it. That way those variables could be referenced in any of our scripts and not have to be set manually in each one. I'll hold off on going that route until @mmlb has chimed in with feedback too.

mmlb commented on .drone.yml:
```diff
@@ -55,6 +55,7 @@ pipeline:
     group: ci1
     image: osie-test-env
     commands:
+      - bash set_version.sh
```
Hmm this should be in the Makefile instead.

mmlb commented on set_version.sh:
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+VERSION=`git rev-parse HEAD`
+BRANCH=`git symbolic-ref --short HEAD`
```
We should add the drone build number too

ScottGarman committed:
Display OSIE version info - osie.sh and osie-runner

Show OSIE version information at the start of osie.sh and in the OSIE
error summary banner, and also the osie-runner motd banner.

Signed-off-by: Scott Garman <sgarman@packet.com>

ScottGarman committed:
Fix file perms when generating initramfs

On Ubuntu 18.04 and 20.04 I discovered that building OSIE locally would
result in a failure because the initramfs would get copied out of the
container as owned by root with user-only read permissions. This updates
the file permissions to ensure the file can be read by the build user.

Signed-off-by: Scott Garman <sgarman@packet.com>
